modules/renter/contractor/downloader.go

```diff
@@ -79,7 +80,12 @@ func (hd *hostDownloader) Sector(root crypto.Hash) ([]byte, error) {
 
 	hd.contractor.mu.Lock()
 	hd.contractor.contracts[contract.ID] = contract
-	hd.contractor.saveSync()
+	cpath := fmt.Sprintf("contracts.%s", contract.ID.String())
+	hd.contractor.persist.update(
+		cpath+".lastrevision", contract.LastRevision,
+		cpath+".lastrevisiontxn", contract.LastRevisionTxn,
+		cpath+".downloadspending", contract.DownloadSpending,
+	)
```

Comment:
The other way we could add safety is by moving the fields themselves into the journal. Then the only way to change them is through the journal, so anything you forget  to updates will start causing bugs immediately instead of being silent until the next restart.